serving/src/main/java/ai/djl/serving/http/LogRequestHandler.java

```diff
@@ -0,0 +1,209 @@
+            throws ModelException {
+
+        File[] files =
+                dir.toFile()
+                        .listFiles(
+                                (f, name) -> {
+                                    if (name.equals(fileName)) return true;
+                                    return false;
+                                });
+        if (files.length == 0) throw new ModelException("File does not exist");
+        FileInputStream input = null;
+        try {
+            input = new FileInputStream(files[0]);
+            int size = (int) files[0].length();
+            byte[] data = new byte[size];
+            if (size > 0) {
+                int offset;
+                int read;
+                for (offset = 0;
+                        offset < size && (read = input.read(data, offset, size - offset)) != -1;
+                        offset += read) {}
+            }
+            HttpResponseStatus status = HttpResponseStatus.OK;
+            FullHttpResponse resp =
+                    new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, status, false);
+            String showName = URLEncoder.encode(fileName, "UTF-8");
+            resp.headers()
+                    .set("Content-Type", "text/plain")
+                    .set(
+                            "Content-Disposition",
+                            "attachment;fileName=" + showName + ";fileName*=UTF-8''" + showName);
+            resp.content().writeBytes(data);
+            NettyUtils.sendHttpResponse(ctx, resp, true);
+        } catch (IOException e) {
+            logger.error("Failed to read log file", e);
+            throw new ModelException("Failed to read log file");
+        } finally {
+            try {
+                input.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private void handleLogList(ChannelHandlerContext ctx, Path dir) {
+        File[] files =
+                dir.toFile()
+                        .listFiles(
+                                (f, name) -> {
+                                    if (name.endsWith(".log")) return true;
+                                    return false;
+                                });
+        List<Map<String, Object>> list =
+                Arrays.stream(files)
+                        .map(
+                                v -> {
+                                    Map<String, Object> m = new HashMap<String, Object>();
+                                    m.put("name", v.getName());
+                                    m.put("lastModified", v.lastModified());
+                                    m.put("length", v.length());
+                                    return m;
+                                })
+                        .collect(Collectors.toList());
+
+        NettyUtils.sendJsonResponse(ctx, list);
+    }
+
+    private void handleLogText(
```

Comment:
```
    private void showLog(ChannelHandlerContext ctx, Path dir, String fileName, int lines) {
        if (fileName.contains("..")) {
            throw new BadRequestException("Invalid log file name:" + fileName);
        }
        Path file = dir.resolve(fileName);
        if (!Files.isRegularFile(file)) {
            throw new BadRequestException("File does not exist");
        }

        String lastLineText = getLastLineText(file.toFile(), lines);
        NettyUtils.sendJsonResponse(ctx, lastLineText);
    }

    private String getLastLineText(File file, int lines) {
        long fileLength = file.length() - 1;
        if (file.length() < 0) {
            return "";
        }

        StringBuilder builder = new StringBuilder();
        try (RandomAccessFile raf = new RandomAccessFile(file, "r")) {
            int readLines = 0;
            raf.seek(fileLength);
            for (long pointer = fileLength; pointer >= 0; pointer--) {
                raf.seek(pointer);
                char c;
                c = (char) raf.read();
                if (c == '\n') {
                    readLines++;
                    if (readLines == lines) break;
                }
                builder.append(c);
                fileLength = fileLength - pointer;
            }
            builder.reverse();
        } catch (IOException e) {
            throw new InternalServerException("Failed to read log file.");
        }
        return builder.toString();
    }
```

Reply:
I moved frontend into plugins/manage-console folder.

Everything is working before you merge the branch. Please take a look and let's discuss offline.